integrations/gen_doc_collector_page.py

```diff
@@ -0,0 +1,75 @@
+import json
+
+"""
+This script reads the integrations/integrations.js file and generates the list of data collection integrations inside collectors/COLLECTORS.md, with proper links that Learn can replace into Learn links.
+"""
+
+# Open integrations/integrations.js and extract the dictionaries
+with open('integrations/integrations.js') as dataFile:
+    data = dataFile.read()
+
+    categories_str = data.split("export const categories = ")[1].split("export const integrations = ")[0]
+    integrations_str = data.split("export const categories = ")[1].split("export const integrations = ")[1]
+
+    categories = json.loads(categories_str)
+    integrations = json.loads(integrations_str)
+
+cat_dict = {}
+data_col_cat = {}
+
+
+def recursive(categories):
+    for category in categories:
+        data_col_cat[category['id']] = category['name']
+        if category['children']:
+            recursive(category['children'])
+
+
+recursive(categories[1]['children'])
+
+
+def construct_dict(array, integration):
+    for element in array:
+        try:
+            cat_dict[element].append(integration)
+        except KeyError:
+            cat_dict[element] = []
+            cat_dict[element].append(integration)
+
+
+md = ""
+
+for integration in integrations:
+    try:
+        construct_dict(integration['meta']['monitored_instance']['categories'], integration)
+    except KeyError:
+        pass
+
+sorted_dict = dict(sorted(cat_dict.items()))
+
+
+for category_id in sorted_dict:
+    heading = '#' * len(category_id.split('.'))
+
+    for cat in data_col_cat:
+        if cat == category_id:
+            name = data_col_cat[cat]
+
+    md += f'#{heading} {name}\n\n\n'
+    names = []
+    for integration in sorted_dict[category_id]:
+        name = integration['meta']['monitored_instance']['name']
+        link = integration['edit_link'].replace("metadata.yaml", "") + \
```

Comment:
The `edit_link` property is generated at runtime by `gen_integrations.py`, it’s not part of the actual `metadata.yaml` files, so you’ll need to generate it yourself here. See https://github.com/netdata/netdata/blob/master/integrations/gen_integrations.py#L385 for the function responsible for this (and note that `_src_path` is also not in the `metadata.yaml` files, and just gets added to the parsed data by the script while parsing them.

Reply:
I don't read metadata.yaml files, I only replace the link and make sure it points to the proper md file

Reply:
Ah, I somehow completely missed that.